cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -263,6 +263,7 @@ private void handleProgramEvent(ProgramRunId programRunId, ProgramRunStatus prog
     RunRecordDetail recordedRunRecord;
     switch (programRunStatus) {
       case STARTING:
+        programLifecycleService.removeInflightRequest(programRunId);
```

Comment:
shall we move it after recordProgramStart ?

Reply:
This will be fixed once [PR](https://github.com/cdapio/cdap/pull/13796) is merged.